src/puppetlabs/puppetdb/utils/metrics.clj

```diff
@@ -21,3 +22,30 @@
   timer objects"
   [timers & body]
   `(multitime!* ~timers (fn [] (do ~@body))))
+
+(defn maybe-prefix-key [prefix key]
+  (if prefix
+    (keyword (str prefix "." (name key)))
+    key))
+
+(defn get-db-name
+  "Returns a string of the db's name or nil. In command broadcast mode the db
+   name is included in the hikari writepool name.
+   e.g. poolname = 'PDBWritePool: foo', db name = 'foo'
+   In normal mode no db specific name is present and function will return nil."
+  [db]
+  ;; when :datasource isn't present in test *db* map assume non-broadcast
+  (some->> db
+           :datasource
+           .getPoolName
+           (re-find #"(?<=PDBWritePool: ).*")))
```

Comment:
Would a change in pool name make this any easier? I'm always scared of regexes, and in addition our current pool name doesn't produce the clearest log messages.
```
2020-06-22 11:54:25,068 INFO  [async-dispatch-2] [c.z.h.HikariDataSource] PDBMigrationsPool: foo - Start completed.
```